### PR TITLE
Use package-private fields to avoid accessor functions

### PR DESCRIPTION
**Pull request by keithc-ca**
* conditionally include `CacheKey.newDeclaredPublicMethodsKey()`
* make some `private` methods `static`

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -240,11 +240,11 @@
 	private static final class MetadataCache {
 		MetadataCache() {}
 
-		private static long cachedCanonicalNameOffset = -1;
-		private static long cachedSimpleNameOffset = -1;
+		static long cachedCanonicalNameOffset;
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/Class.java:243]: Won't removing the initialization to -1 break the code by returning 0 for the offset?
```
		MetadataCache.cachedSimpleNameOffset = getFieldOffset(
				MetadataCache.class, "cachedSimpleName", MetadataCache.cachedSimpleNameOffset);
```
```
private long getFieldOffset(Class<?> hostClass, String fieldName, long initialOffset) {
	if (initialOffset == -1) {
...
	}
	return initialOffset;
```

Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/Class.java:243]: Yes, you're right. I'll restore that initialization.

**Comment by keithc-ca** [comment]
For a successful travis build, see https://travis-ci.org/github/keithc-ca/openj9/builds/739672022.

**Review by pshipton — APPROVED** [review]

**Comment by keithc-ca** [comment]
For the record, the comment referring to the travis build was meant for #10999.

**Pull request merged**